### DISCUSSION TITLE
Error "An error occurred (ModelError) when calling the InvokeEndpoint operation"

### DISCUSSION THREAD

**Discussion by sergioCancanEs** [aws/amazon-sagemaker-examples#4747] [Q&A]
# Error 500 when consuming SageMaker endpoint deployed with MLflow

## Description
Hello everyone,
I successfully replicated the notebook "sm-mlflow_deployment.ipynb" in my JupyterLab Space. However, when I try to consume the endpoint from another notebook or with the SageMaker endpoint test inference tool, I consistently receive a 500 error related to the model.

## Steps to Reproduce
1. Deploy the model using the "sm-mlflow_deployment.ipynb" notebook.
2. Attempt to consume the endpoint using the following code:

```python
import sagemaker
import boto3
import numpy as np

sagemaker_session = sagemaker.Session()
endpoint_name = 'endpoint_name'

# SageMaker Runtime client
client = boto3.client('sagemaker-runtime')

payload = bytes(np.array([1.0, 2.0, 3.0, 4.0]).reshape(1,-1))

response = client.invoke_endpoint(
    EndpointName=endpoint_name,
    ContentType='application/json',  
    Body=payload
)

print(response)

```
Error
```python
ModelError: An error occurred (ModelError) when calling the InvokeEndpoint operation: Received server error (500) from primary with message "<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>

```
Questions

How can I consume an endpoint deployed from SageMaker with MLflow in another notebook or service (e.g., Lambda)?
What could be causing this 500 error, and how can I troubleshoot it?

**Comment by rahulmisal27** [comment]
@sergioCancanEs Make sure your deployed endpoint is healthy as SageMaker checks for health first and then make sure your lambda function has sagemaker: invoke endpoint permission added to the IAM role attached to it.
